src/readers/morphologyHDF5.h

```diff
@@ -12,6 +12,92 @@ namespace h5 {
 Property::Properties load(const std::string& uri);
 Property::Properties load(const HighFive::Group& group);
 
+namespace detail {
+/**
+ * Provides the API for reading datasets from HDF5 files.
+ *
+ * The CRTP requirements are that `Derived` has a method
```

Comment:
do we need to CRTP vs virtual, and rely on devirtualization? Is this a hot dispatch?

Reply:
I think we can't use virtual function calls, because of the `T& out`. This is usually `std::vector<T>& out` but the inner type differs. I don't think this goes away even if we were to switch to `T read(...) const`. Therefore, the only way I know to avoid a bit of code duplication is to use CRTP.